commit f9555d18084985c80a91baa4fdb7d205b401a754
Author: Bairavi Alagappan <bairavix.alagappan@intel.com>
Date:   Fri Mar 14 13:14:29 2025 +0000

    crypto: qat - set parity error mask for qat_420xx
    
    The field parerr_wat_wcp_mask in the structure adf_dev_err_mask enables
    the detection and reporting of parity errors for the wireless cipher and
    wireless authentication accelerators.
    
    Set the parerr_wat_wcp_mask field, which was inadvertently omitted
    during the initial enablement of the qat_420xx driver, to ensure that
    parity errors are enabled for those accelerators.
    
    In addition, fix the string used to report such errors that was
    inadvertently set to "ath_cph" (authentication and cipher).
    
    Fixes: fcf60f4bcf54 ("crypto: qat - add support for 420xx devices")
    Signed-off-by: Bairavi Alagappan <bairavix.alagappan@intel.com>
    Signed-off-by: Giovanni Cabiddu <giovanni.cabiddu@intel.com>
    Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>

diff --git a/drivers/crypto/intel/qat/qat_420xx/adf_420xx_hw_data.c b/drivers/crypto/intel/qat/qat_420xx/adf_420xx_hw_data.c
index 98f3c1210666..4feeef83f7a3 100644
--- a/drivers/crypto/intel/qat/qat_420xx/adf_420xx_hw_data.c
+++ b/drivers/crypto/intel/qat/qat_420xx/adf_420xx_hw_data.c
@@ -411,6 +411,7 @@ static void adf_gen4_set_err_mask(struct adf_dev_err_mask *dev_err_mask)
 	dev_err_mask->parerr_cpr_xlt_mask = ADF_420XX_PARITYERRORMASK_CPR_XLT_MASK;
 	dev_err_mask->parerr_dcpr_ucs_mask = ADF_420XX_PARITYERRORMASK_DCPR_UCS_MASK;
 	dev_err_mask->parerr_pke_mask = ADF_420XX_PARITYERRORMASK_PKE_MASK;
+	dev_err_mask->parerr_wat_wcp_mask = ADF_420XX_PARITYERRORMASK_WAT_WCP_MASK;
 	dev_err_mask->ssmfeatren_mask = ADF_420XX_SSMFEATREN_MASK;
 }